components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/TextInputView.kt

```diff
@@ -0,0 +1,166 @@
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    singleLine: Boolean = false,
+    characterCount: Int? = null,
+    prefix: @Composable (() -> Unit)? = null,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default
+) {
+
+    var localValue: String by rememberSaveable { mutableStateOf("") }
+
+    var localError: String? by rememberSaveable { mutableStateOf(null) }
+
+    localError = errorText
+
+    val counterEnabled: Boolean = characterCount != null
+
+    fun errorText(): @Composable (() -> Unit)? {
+        val text: String? = errorText
+        return text?.let {
+            {
+                Text(
+                    text = text,
+                    modifier = Modifier.semantics {
+                        if (errorContentDescription != null) {
+                            contentDescription = errorContentDescription
+                        }
+                    }
+                )
+            }
+        }
+    }
+
+    @Composable
+    fun errorTextCounterCombo(): @Composable (() -> Unit) {
+        return {
+            Row {
+                Column(horizontalAlignment = Alignment.Start, modifier = Modifier.fillMaxWidth(0.8f)) {
+                    if (errorText != null) {
+                        Text(
+                            text = errorText,
+                            modifier = Modifier.semantics {
+                                if (errorContentDescription != null) {
+                                    contentDescription = errorContentDescription
+                                }
+                            }
+                        )
+                    }
+                }
+                Column(horizontalAlignment = Alignment.End, modifier = Modifier.fillMaxWidth(0.95f)) {
+                    Text(
+                        text = "${localValue.length}/$characterCount",
+                        textAlign = TextAlign.End,
+                    )
+                }
+            }
+        }
+    }
+
+    fun modifier(): Modifier {
+        return if (counterEnabled) {
+            modifier.fillMaxWidth().padding(top = hmrc_icon_size_24).padding(horizontal = hmrc_spacing_16)
+        } else if (!localError.isNullOrEmpty()) {
+            modifier.fillMaxWidth().padding(top = hmrc_icon_size_24).padding(horizontal = hmrc_spacing_16)
+        } else {
+            modifier.fillMaxWidth().padding(vertical = hmrc_icon_size_24).padding(horizontal = hmrc_spacing_16)
+        }
+    }
+
+    OutlinedTextField(
+        modifier = modifier(),
+        isError = !localError.isNullOrEmpty() || (localValue.length > (characterCount ?: Int.MAX_VALUE)),
+        value = localValue,
+        onValueChange = {
+            if (onInputValueChange != null) {
+                onInputValueChange(it)
+            }
+            localValue = if (inputFilter != null && it != "") {
```

Comment:
There's a kotlin extension we can use for the empty check. Does the same thing, but slightly "more kotlin".

```suggestion
            localValue = if (inputFilter != null && it.isNotEmpty()) {
```